src/stories/Tooltip.tsx

```diff
@@ -0,0 +1,108 @@
+import {useState, useRef, useEffect, ReactNode} from "react";
+import {motion, AnimatePresence} from "framer-motion"
+
+//content
+//customizable positions
+//triggers on button hover
+//delays
+
+interface TooltipProps {
+  children: ReactNode;
+  content: string;
+  position: 'top' | 'bottom' | 'left' | 'right';
+}
+
+function Tooltip({children = <button>Hover Me</button>, content, position}: TooltipProps) {
+  const [isVisible, setIsVisible] = useState(false)
+  const [shouldRender, setShouldRender] = useState(false)
+  const wrapperRef = useRef<HTMLDivElement | null> (null)
+
+  useEffect(() => {
+
+    const triggerElement = wrapperRef.current
+  
+    const showTooltip = () => {
+      setShouldRender(true)
+      setTimeout(()=> 
+        setIsVisible(true), 300)
+    };
+
+    const hideTooltip = () => {
+      setIsVisible(false)
+      setTimeout(()=> setShouldRender(false), 200) 
+    };
+  
+    if (triggerElement) {
+      // hover start
+      triggerElement.addEventListener('mouseenter', showTooltip);
+      // hover end
+      triggerElement.addEventListener('mouseleave', hideTooltip);
+
+      //focus events
+      triggerElement.addEventListener('focus', showTooltip);
+      triggerElement.addEventListener('blur', hideTooltip);
+    }
+    return () => {
+      if (triggerElement) {
+        //hover
+        triggerElement.removeEventListener('mouseenter', showTooltip);
+        triggerElement.removeEventListener('mouseleave', hideTooltip);
+
+        //focus 
+        triggerElement.removeEventListener('focus', showTooltip);
+        triggerElement.removeEventListener('blur', hideTooltip);
+      }
+    };
+  }, [])
+
+  //for conditional styling depending on position
+  function getPositionClasses(position: 'top' | 'bottom' | 'left' | 'right'): string {
```

Comment:
pull outside of the render loop

Redefining "position" in multiple places - seems like a good opportunity to pull it out to a generic type or enum : )